pkg/kv/kvserver/rangefeed/registry.go

```diff
@@ -146,6 +136,7 @@ func newRegistration(
 	}
 	r.mu.Locker = &syncutil.Mutex{}
 	r.mu.caughtUp = true
+	r.mu.catchUpIter = catchUpIter
```

Comment:
The comment for this fields needs updating: it's now constructed on the rangefeed server goroutine, before being passed to the processor.